### ISSUE TITLE
Mechanism for requesting reads in GA4GH binary format

### ISSUE THREAD

**Issue by jmarshall** [samtools/hts-specs#596]
We may wish to add a mechanism for requesting reads in other GA4GH binary formats, e.g., `fmt=proto` for protocol buffers.

Suggestion by dglazer. Further musing: How will compression work in this case — can we benefit from columnar compression as does Parquet?

[This issue was previously listed as a “possible future enhancement” within the htsget specification text itself.]

**Comment by lbergelson** [comment]
It's fine if people want to serve reads in different formats, but ideally we would have a spec for the exact protobuff format.  The previous (and admittedly ancient and probably intended more as a proof of concept than as a production system) google custom format we worked with was underspecified and hard to consume because it didn't exactly match the SamRecord data model and left a lot of ambiguity in how you were meant to consume the tags. 

If we want to serve column compressed reads, I've heard of a format called CRAM which is optimized for exactly that use case.

**Comment by jmarshall** [comment]
Indeed. This issue is a record of a early footnote that has now been removed from the spec itself — and which has probably been superseded by the direction that htsget and other events have taken, and probably can just be closed.

**Comment by brainstorm** [comment]
There are other binary formats that start to ask for htsget support, actually: https://github.com/igvteam/igv/issues/1027#issuecomment-922498253

**Comment by jmarshall** [comment]
That IGV issue is about interval / quantitative interval formats such as BED, BigWig, and in particular D4.

This issue is about additional formats for reads data (and is mostly historical). BED/BigWig/D4 would be a separate category of data and separate endpoint alongside `/reads` and `/variants`, which is a larger question unrelated to this issue.

**Comment by jkbonfield** [comment]
CRAM may be a bit heavy-weight for some applications, plus I can also see a desire for wanting a referenceless data stream.  This is obviously possible in CRAM, but involves quite a lot of transcoding at the server end to go from reference-based to reference-less encoding.

I don't know enough about Protobuff, but my very limited understanding is it's basically a binary serialisation process.  Is that so?  If so it's more akin to BAM than CRAM and columnar compression isn't possible.  I'd expect protobuff + zstd though could be viable, certainly as an alternative for BAM consumption.

There are also other formats which are almost certainly a direct drop-in here, such as Apache Arrow.

**Comment by jmarshall** [comment]
This issue records a “possible future enhancement” footnote that was added to the htsget spec prior to October 2016. I paraphrased from the original which said “request reads in GA4GH binary format”, which as @lbergelson noted was a reference to the now long-since retired [ga4gh-schemas protobuf representation](https://github.com/ga4gh/ga4gh-schemas/tree/master/src/main/proto/ga4gh).

As that particular binary format never reached deployment and is no longer of practical interest, as discussed in our most recent meeting we'll close this issue.
